### DISCUSSION TITLE
Understanding caching

### DISCUSSION THREAD

**Discussion by ATakaSKY** [vercel/next.js#47732] [Help]
### Summary

I'm playing with SSG, ISR aspects of next and confused in caching bit at the moment.

My question is what is the advantage of using such workflow if I use a CDN, for example AWS Cloudfront? Even if the page is built dynamically per each request in (as in getServerSideProps) the page will still get cached by Cloudfront and returned very fast. So I'm wondering what is the real-world use case of getStaticProps because any serious website already uses CDN. I understand that the technology is primarily geared towards CMS websites but even in those content may change periodically on the same resource (e.g. an article had a grammar mistake and it must be corrected) and it's an advantage that the page can change dynamically not per build time.

**Comment by dalhaan** [comment]
I think the main advantage is cost and consistency. CDNs normally only cache files at one edge location at a time, so if someone visits your site from New Zealand, the response will only be cached at the edge location closest to them, not globally. If someone visits your site from another country, they will most likely go through a different edge location where it may not be cached, resulting in your server being hit directly again. Having an SSG/ISR cache (in an S3 bucket for instance) is probably cheaper (for the person paying the server bills) and faster (for the user) as you don't have to rebuild the same HTML & page data JSON for every CDN miss, especially if you are hitting external APIs that cost money to get the data for the page.
